internal/kzg/kzg_prove.go

```diff
@@ -1,161 +0,0 @@
-package kzg
-
-import (
-	"github.com/consensys/gnark-crypto/ecc/bls12-381/fr"
-)
-
-// Create a KZG proof that a polynomial f(x) when evaluated at a point `z` is equal to `f(z)`
-//
-// [compute_kzg_proof_impl](https://github.com/ethereum/consensus-specs/blob/3a2304981a3b820a22b518fe4859f4bba0ebc83b/specs/deneb/polynomial-commitments.md#compute_kzg_proof_impl)
-func Open(domain *Domain, p Polynomial, point fr.Element, ck *CommitKey) (OpeningProof, error) {
-	if len(p) == 0 || len(p) > len(ck.G1) {
```

Comment:
It's interesting that we allow `len(p) < len(ck.G1)` here. IIUC given the API here, if someone wanted to commit to a low degree polynomial they would also have to provide a custom domain with a small subgroup. Is this something useful for protodank? Or should we expect that users would just pad the polynomial with zeroes or something?